### PR TITLE
Redirect the detail view after modifying a cluster

### PR DESCRIPTION
**Pull request by maurodoglio**
This is based on https://github.com/mozilla/telemetry-analysis-service/pull/20

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-analysis-service/21)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: atmo/clusters/urls.py [hunk 1]
```diff
@@ -5,4 +5,5 @@
     url(r'^new/$', views.new_cluster, name='clusters-new'),
     url(r'^edit/$', views.edit_cluster, name='clusters-edit'),
     url(r'^delete/$', views.delete_cluster, name='clusters-delete'),
+    url(r'^(?P<id>[0-9]+)/$', views.detail_cluster, name='clusters-detail'),
```
Comment (jezdez) [inline atmo/clusters/urls.py]: `\d` will do instead of `[0-9]` right?

File: atmo/static/css/base.css [hunk 2]
```diff
@@ -10,3 +10,7 @@
   -ms-user-select: none;
   user-select: none;
 }
+
+div.container.main {
+  padding-top:50px;
+}
```
Comment (jezdez) [inline atmo/static/css/base.css]: linebreak EOL?

File: atmo/templates/atmo/detail-cluster.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {% endif %}
+  </div>
+</div>
+{% if cluster.is_active %}
+  <div class="row">
+    <div class="col-md-2"><strong>Terminates at:</strong></div>
+    <div class="col-md-10">{{cluster.end_date}}</div>
+  </div>
+  <p>
+    <form action="/clusters/delete/" method="POST" enctype="multipart/form-data">
```
Comment (jezdez) [inline atmo/templates/atmo/detail-cluster.html]: Let's use `{% url 'clusters-delete' %}` here

File: atmo/templates/atmo/detail-cluster.html [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      {{ cluster.master_address }}
+    {% else %}
+    Please wait, instance details will appear when the master is ready.
+    {% endif %}
+  </div>
+</div>
+{% if cluster.is_active %}
+  <div class="row">
+    <div class="col-md-2"><strong>Terminates at:</strong></div>
+    <div class="col-md-10">{{cluster.end_date}}</div>
```
Comment (jezdez) [inline atmo/templates/atmo/detail-cluster.html]: whitespace around curly braces

File: atmo/templates/atmo/detail-cluster.html [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  </div>
+  <p>
+    <form action="/clusters/delete/" method="POST" enctype="multipart/form-data">
+      {% csrf_token %}
+      <input name="cluster" value="{{cluster.id}}" type="hidden">
```
Comment (jezdez) [inline atmo/templates/atmo/detail-cluster.html]: whitespace

File: atmo/templates/atmo/detail-cluster.html [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      <code>ssh -i my-private-key -L 8888:localhost:8888 hadoop@{{ cluster.master_address }}</code>
+    </p>
+    <p>
+      You can save some typing by adding defaults to your <code>~/.ssh/config</code> file:
+    </p>
+    <pre>Host *.compute.amazonaws.com
+  User hadoop
+  LocalForward 8888 localhost:8888
+  Identityfile ~/.ssh/{my-private-key}</pre>
+    To upload an existing notebook, use SCP:
```
Comment (jezdez) [inline atmo/templates/atmo/detail-cluster.html]: Let's also add a link here `ssh://hadoop@{{ cluster.master_address }}` here, so once people have added the section to `~/.ssh/config` they can just click on it to launch their terminal

File: atmo/clusters/models.py [hunk 7]
```diff
@@ -54,7 +55,7 @@ class Cluster(models.Model):
         help_text="Most recently retrieved AWS status for the cluster."
     )
     master_address = models.CharField(
-        max_length=255, default="",
+        max_length=255, default="", blank=True,
```
Comment (jezdez) [inline atmo/clusters/models.py]: I *think* this requires a new migration since it tracks all parameters, right?

File: atmo/clusters/models.py [hunk 8]
```diff
@@ -121,3 +122,6 @@ def is_active(self):
     @property
     def is_ready(self):
         return self.most_recent_status == 'WAITING'
+
+    def get_absolute_url(self):
+        return reverse('clusters-detail', kwargs={'id': self.id})
```
Comment (jezdez) [inline atmo/clusters/models.py]: linebreak EOL

File: atmo/views.py [hunk 9]
```diff
@@ -17,7 +18,9 @@
 @login_required
 def dashboard(request):
     username = request.user.email.split("@")[0]
-    clusters = Cluster.objects.filter(created_by=request.user).order_by("start_date")
+    clusters = (Cluster.objects.filter(created_by=request.user)
+                .filter(end_date__gt=timezone.now()-timedelta(days=1))
```
Comment (jezdez) [inline atmo/views.py]: minor cosmetic thing, indent the filter calls to the first one

File: atmo/clusters/urls.py (hunk 1 above)
Comment (maurodoglio) [inline atmo/clusters/urls.py]: 👍

File: atmo/static/css/base.css (hunk 2 above)
Comment (maurodoglio) [inline atmo/static/css/base.css]: 👍

Comment (maurodoglio) [inline atmo/static/css/base.css]: 👍

File: atmo/templates/atmo/detail-cluster.html (hunk 4 above)
Comment (maurodoglio) [inline atmo/templates/atmo/detail-cluster.html]: :+1:

File: atmo/templates/atmo/detail-cluster.html (hunk 3 above)
Comment (maurodoglio) [inline atmo/templates/atmo/detail-cluster.html]: 👍

File: atmo/templates/atmo/detail-cluster.html (hunk 5 above)
Comment (maurodoglio) [inline atmo/templates/atmo/detail-cluster.html]: 👍

File: atmo/clusters/models.py (hunk 7 above)
Comment (maurodoglio) [inline atmo/clusters/models.py]: right you are

File: atmo/templates/atmo/detail-cluster.html (hunk 6 above)
Comment (maurodoglio) [inline atmo/templates/atmo/detail-cluster.html]: :+1:

File: atmo/clusters/models.py (hunk 8 above)
Comment (maurodoglio) [inline atmo/clusters/models.py]: 👍

File: atmo/views.py (hunk 9 above)
Comment (maurodoglio) [inline atmo/views.py]: 👍

**Pull request merged**